### ISSUE TITLE
Make the border button hover use circles

### ISSUE THREAD

**Issue by Jacalz** [FyshOS/fynedesk#238]
### Is your feature request related to a problem? Please describe:

I think the current rectangles used when hovering look a bit too square and clunky. My opinion is that changing them to use circles would result in a slightly more pleasing visual experience.

### Is it possible to construct a solution with the existing API?

Yes. The red close button hover rectangle is the easiest to switch but the other buttons might be more problematic as it is the standard hover shadow of the widget.Button.

### Describe the solution you'd like to see:

Basically just masking sure that hovering the close, maximize and minimize buttons shows a circle instead of a rectangle. I had a very quick test with just switching the red close button hover to a circle with quite good results but the circle was just a slight bit too big.

My idea is to have the buttons look like this on hover (but just for the icon being hovered of course):
![image](https://user-images.githubusercontent.com/25466657/211144097-c5cf7791-98ec-49be-9e8c-5d67619ca7ef.png)
That images is a libadwaita app so it technically always shows the circle behind the buttons but I think you get the idea (had troubles getting screenshots working reliably for various reasons).

**Comment by andydotxyz** [comment]
The buttons are currently buttons. Moving to a circle and we'd need to build custom widgets that are specific to borders.

Just noting that for anyone who comes in and thinks it could be a quick fix to pick up.

**Comment by Jacalz** [comment]
Indeed. I would personally not be against having round hover indication for buttons in Fyne upstream. It looks better in my opinion but I guess it depends on what kind of icon is used.

**Comment by andydotxyz** [comment]
> Indeed. I would personally not be against having round hover indication for buttons in Fyne upstream. It looks better in my opinion but I guess it depends on what kind of icon is used.

I don't think we can do that for all buttons - a circle around text really doesn't make sense.

**Comment by Jacalz** [comment]
> > Indeed. I would personally not be against having round hover indication for buttons in Fyne upstream. It looks better in my opinion but I guess it depends on what kind of icon is used.
> 
> I don't think we can do that for all buttons - a circle around text really doesn't make sense.

That's very true. I probably should have clarified that I talked about low importance buttons without text and only an icon :sweat_smile:

**Comment by andydotxyz** [comment]
I think we might have to make icons smaller to support that... which makes them no longer the standard icon size.
I could be wrong as I didn't mock it up - but it seems like it would be tough to fit it all in.

**Comment by rafaelsousa** [comment]
Isn't the round effect of the buttons just round corners on a rectangle? I mean, the icon is small and there is no upper and down corners.

**Comment by rafaelsousa** [comment]
IMHO, I think in order to do this, we should have an rounded_rectangle or have an extra property **(Radius)** in the Rectangle struct in order to implement this feature.

**Comment by andydotxyz** [comment]
Rectangle has `Radius` on `develop` branch now :)
